distributed/worker.py

```diff
@@ -4576,3 +4591,74 @@ def warn(*args, **kwargs):
         worker.log_event("warn", {"args": args, "kwargs": kwargs})
 
     warnings.warn(*args, **kwargs)
+
+
+def benchmark_disk(
+    sizes=["1 kiB", "100 kiB", "1 MiB", "10 MiB", "100 MiB"],
+    rootdir=None,
+    duration=1.0,
+) -> dict:
+    duration = parse_timedelta(duration)
+
+    out = {}
+    for size_str in sizes:
+        with tmpdir(dir=rootdir) as dir:
+            dir = pathlib.Path(dir)
+            names = list(map(str, range(100)))
+            size = parse_bytes(size_str)
+
+            data = os.urandom(size)
+
+            start = time()
+            total = 0
+            while time() < start + duration:
+                with open(dir / random.choice(names), mode="ab") as f:
+                    f.write(data)
+                    os.fsync(f)
+                total += size
+
+            out[size_str] = total / (time() - start)
+    return out
+
+
+def benchmark_memory(
+    sizes=["2 kiB", "10 kiB", "100 kiB", "1 MiB", "10 MiB"],
+    duration=0.2,
+) -> dict:
+    duration = parse_timedelta(duration)
+    out = {}
+    for size_str in sizes:
+        size = parse_bytes(size_str)
+        data = os.urandom(size)
+
+        start = time()
+        total = 0
+        while time() < start + duration:
+            data[:-1]
+            total += size
+
+        out[size_str] = total / (time() - start)
+    return out
+
+
+async def benchmark_network(
+    sizes=["1 kiB", "10kiB", "100kiB", "1 MiB", "10 MiB", "50 MiB"],
+    rpc=rpc,
+    duration=1.0,
+    address=None,
```

Comment:
this parameter is mandatory

Reply:
I gather from your comment above that you don't like adding `=None` for a mandatory parameter.  Unfortunately I wasn't able to understand what a good resolution for this would be.  If you can point me in a good direction here that would help.

Reply:
just ``address: str`` and move it to the top of the list of parameters

Reply:
Done